File: server/app/graphql/mutations/image/create_listing_image.rb
class Mutations::Image::CreateListingImage < Mutations::BaseMutation
  null false

  argument :uuid, ID, required: true
  argument :file, Types::FileType, required: false

  field :listing, ObjectTypes::Listing, null: true
  field :error, String, null: true

  def resolve(uuid:, file:)
    listing = ::Listing.find_by(uuid:)
    listing.images.create!(image: file)

    { listing: listing.reload }
  rescue => e
    return GraphQL::ExecutionError.new(e.message)
  end
end

File: storeadmin/api/mutations/image/createListingImage.ts
import { imageFragment, listingFragment } from '@/api/fragments'
import gql from 'graphql-tag'

export default gql`
  mutation createListingImage($uuid: ID!, $file: File!) {
    createListingImage(input: { uuid: $uuid, file: $file }) {
      listing {
        ...listingFragment
        images {
          ...imageFragment
        }
      }
      error
    }
  }
  ${imageFragment}
  ${listingFragment}
`
